api/controllers/groups/groupEvents.ts

```diff
@@ -42,3 +43,54 @@ export const groupJoinEvent = async (
     next(err);
   }
 };
+
+export const groupLeaveEvent = async (
+  req: Request<{}, getResponsesBody<JoinEvent>, getRequestBody<JoinEvent>, {}>,
+  res: core.Response<getResponsesBody<JoinEvent>, {}, getHTTPCode<JoinEvent>>,
+  next: NextFunction,
+): Promise<
+  core.Response<getResponsesBody<JoinEvent>, {}, getHTTPCode<JoinEvent>>
+> => {
+  try {
+    const group = await getRepository(Groups).findOne({
+      where: { id: req.body.groupId },
+      relations: ['events'],
+    });
+
+    if (!group) {
+      return res.status(404).json({ msg: 'E_GROUP_DOES_NOT_EXIST' });
+    }
+
+    const eventToLeave = await getRepository(Event).findOne({
+      id: req.body.eventId,
+    });
+
+    if (!eventToLeave) {
+      return res.status(404).json({ msg: 'E_EVENT_DOES_NOT_EXIST' });
+    }
+
+    //searching for the event in the list of the events already joined
+    let found = false;
+    let index: number;
+
+    for(let e of group.events) {
+      if (e === eventToLeave){
+        found = true;
+        index = group.events.indexOf(e);
+        break;
+      }
+    }
```

Comment:
Could use [array.indexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) instead, it will be more clear